src/boot/efi/boot.c

```diff
@@ -727,9 +727,10 @@ static BOOLEAN menu_run(
                         ST->ConOut->OutputString(ST->ConOut, clearline + 1 + x + len);
                 }
 
-                if (do_beep) {
+                /* Beep several times so that the selected entry can be distinguished. */
+                for (UINTN i = 0; config->beep && i <= idx_highlight; i++) {
                         beep();
-                        do_beep = FALSE;
+                        BS->Stall(400 * 1000);
```

Comment:
Maybe move this inside `beep()` so that there's no stall on !x86?

Reply:
Oops. Totally slipped my mind.